### ISSUE TITLE
Depending on which of the series go first, tooltip won't show for part of the data

### ISSUE THREAD

**Issue by mmiszy** [n3-charts/line-chart#228]
If there are series of data with offset (eg. some data has `null` or `undefined` where other have values) then depending on which of the series is listed first in the `options.series` array, tooltip won't work for part of the data.

Example: http://codepen.io/mmiszy/pen/bdwxGv?editors=101

Hover over the graph. Tooltip won't show for the green orange on the right. If you change order of the elements in `graph.options.series` then tooltip won't work for the graph on the left.

**Comment by chaosmail** [comment]
Hi @mmiszy 
thanks for reporting the issue. It is definitely an error and we need to take a closer look - personally I also don't like too much that the scrubber keeps appearing for the first/last value of the other datasets. I will keep you updated update that. Btw. how are you organizing your datasets? Are you using multiple arrays and merge them together in order to display them?
Best, Christoph

**Comment by mmiszy** [comment]
HI @chaosmail, thanks for getting back to me.
In our case it's even more complicated than using multiple arrays and merging them together.

Our data is always plotted against time – we're supposed to have a value every minute in the last 24 hours. But sometimes parts of the data are missing (imagine we're plotting readings from servers but sometimes some of them are offline). Due of that we don't always have 60 * 24 = 1440 values to plot, but less. Moreover there are different series (readings from different servers which sometimes are offline at different periods of time).

Thus what we do is:

1. prepare an array `graphData` with just `x` values – in a loop generate `Date` objects for each minute in the last 24 hours
2. get the first set of data and find where on the graph it should be plotted (sometimes we discard part of the data which is old, sometimes we start the plot somewhere in the middle of the `x` axis
3. iterate over the data and add items to the `graphData` array at appropriate indexes (sometimes there are holes in the data, eg. one reading at 8:31pm and another at 11:59pm instead of 8:32pm)
4. repeat steps 2-3 for each of the selected sets of data

I can imagine some of these steps could be solved by d3/n3 but I don't expected that to implemented any time soon :smile: Unfortunately due to character of the date I work with I've discovered the bug described in the first post and it's quite a blocker for the project… :cry:

**Comment by chaosmail** [comment]
Hi @mmiszy 
thanks for the nice description of your dataset. I will get back to you about this some other day, because we need to investigate first how we can make data loading easier - both in general and in your case.
For the other problem that you mentioned I already sent a PR, so it shouldn't block your project anymore once it is merged into the master.
Best,
Christoph

**Comment by mmiszy** [comment]
Wow :) This is absolutely the best ”customer support” I have ever experienced in an Open Source project. Thank you so much for your help :+1:

**Comment by chaosmail** [comment]
haha thanks :) however, when you submit an issue to D3.js itself, it gets fixed and merged in a couple of seconds
